agent/agentserver/server.go

```diff
@@ -146,6 +155,24 @@ func (s *Server) deleteBlobHandler(w http.ResponseWriter, r *http.Request) error
 	return nil
 }
 
+// preloadTagHandler triggers docker daemon to download specified docker image.
+func (s *Server) preloadTagHandler(w http.ResponseWriter, r *http.Request) error {
+	tag, err := httputil.ParseParam(r, "tag")
+	if err != nil {
+		return err
+	}
+	parts := strings.Split(tag, ":")
+	if len(parts) != 2 {
+		return handler.Errorf("failed to parse docker image tag")
+	}
+	if err := s.dockerCli.ImagePull(
+		context.Background(), r.Host, parts[0], parts[1]); err != nil {
+
+		return handler.Errorf("failed to trigger docker pull: %s", err)
```

Comment:
Nit: avoid 'failed to' prefix, per styleguide: https://github.com/uber-go/guide/blob/master/style.md#error-wrapping